# bulk-RNAseq

## Introduction

This repo contains in-house tutorials and outside links primarily related to bulk RNA-seqencing data processing and analysis. Basic pipelines and default options are primarily used throughout the tutorials. Additional links to package and software documentation are provided in each tutorial - please be sure to review  these documents when completing your own analysis. A separate repo for sc/sn-RNAseq will be developed in the future. 

## Getting Started

The Allen Lab has a Server dedicated to bioinformatics - this server can be used to store and process your sequencing data. Backups of all raw data generated in the lab should be kept on the server. For information on how to access the server or for a brief training on navigating the linux command line, please reach out to JB or LLB. If you experience any issues on the lab server, please reach out to JB or LLB asap. 

R/Rstudio can be downloaded and used locally on your computer or through the lab server. For help accessing or downloading R/Rstudio, feel free to reach out to JB. 


## In-house tutorials/notes

| Link | Description |
| ----------- | ----------- |
|[R 101 Part 1](/R101_Part1.html) | Getting Started with R |
| [Synapse Quantification & Stats Tutorial](/SynapseQuant_Statistics.html) | Basic R tutorial for graphing and statistics |
| [RNAseq Pipeline w/ Salmon + DESeq2](/matt_aging_astrocyte_tutorial.html) | RNA-seq analysis in R after Salmon quasi-mapping |
|[PDF: RNAseq Data Processing STAR & FeatureCounts](/RNAseq_STARFeatureCounts_Tau12mRibotagISH.pdf) | Command line notes/scripts for RNA-seq data processing with STAR and FeatureCounts using ISH data |
|[HTML: RNAseq Data Processing STAR & FeatureCounts](/RNAseq_STARFeatureCounts_Tau12mRibotagISH.html) | HTML version of the above notes | 

## Below are some additional resources that I've found particularly useful

### Linux Command Line

[Cheatsheet 1](https://phoenixnap.com/kb/linux-commands-cheat-sheet#linux-commands-cheat-sheet-pdf)

[Cheatsheet 2](https://www.guru99.com/linux-commands-cheat-sheet.html)

### Salmon for RNA-seq

[Official Salmon Docs](https://salmon.readthedocs.io/en/latest/salmon.html)

[Example workflow with good explanations](https://hbctraining.github.io/Intro-to-rnaseq-hpc-O2/lessons/08_salmon.html)
 
### General RNA-seq pipelines

[Comparison of methods](https://hbctraining.github.io/Training-modules/planning_successful_rnaseq/slides/RNAseq-strategies_mm.pdf)

### R Studio

[Cheatsheets](https://www.rstudio.com/resources/cheatsheets/)

### Markdown

[Offical Markdown Guide](https://www.markdownguide.org/basic-syntax/)

#### 
Reach out to Jillybeth if you have suggestions, questions or comments related to the information provided above.

